dev/tasks/conda-recipes/r-arrow/build.sh

```diff
@@ -14,3 +14,5 @@ fi
 
 # ${R_ARGS} necessary to support cross-compilation
 ${R} CMD INSTALL --build r/. ${R_ARGS}
+# Ensure that features are enabled in the R build (feel free to add others)
+${R} -s -e 'library(arrow); stopifnot(arrow_with_dataset(), arrow_with_parquet(), arrow_with_s3())'
```

Comment:
@h-vetinari Could you review conda part?

Reply:
That part is fine and already working (as in: shows the build fails) in https://github.com/conda-forge/r-arrow-feedstock/pull/63.

However, I don't see how the BREW-specific patch above would apply to conda-forge (we'd need `$PREFIX`), nor do I know what changed between 10 & 11. While taking a first look, my comment in that conda-forge PR was:
> Do we need to do something like in the upstream R [Makefile](https://github.com/apache/arrow/blob/apache-arrow-11.0.0/r/Makefile#L37-L38) to discover the deps correctly?